src/connection.js

```diff
@@ -663,6 +717,12 @@ class Connection extends EventEmitter {
     if (this.state === this.STATE.REROUTING) {
       this.debug.log('Rerouting to ' + this.routingData.server + ':' + this.routingData.port);
       return this.dispatchEvent('reconnect');
+    } else if (this.state === this.STATE.TRANSIENT_FAILURE_RETRY) {
+      const server = this.routingData ? this.routingData.server : this.server;
+      const port = this.routingData ? this.routingData.port : this.config.options.port;
+      this.debug.log('Retry after transient failure connecting to ' + server + ':' + port);
+
+      return this.dispatchEvent('retry');
```

Comment:
I think that'd make it harder to follow the logic. Now it's easy to see what's happening by looking at code under <code>TRANSIENT_FAILURE_RETRY</code> and back track from there.